sys/newlib/syscalls.c

```diff
@@ -85,15 +85,14 @@ void _exit(int n)
  *
  * @return [description]
  */
-caddr_t _sbrk_r(struct _reent *r, ptrdiff_t incr)
+void *_sbrk_r(struct _reent *r, ptrdiff_t incr)
 {
     unsigned int state = disableIRQ();
-    caddr_t res = heap_top;
+    void *res = heap_top;
 
-    if (((incr > 0) && ((heap_top + incr > &_eheap) || (heap_top + incr < res))) ||
-        ((incr < 0) && ((heap_top + incr < &_sheap) || (heap_top + incr > res)))) {
```

Comment:
But wasn't `heap_top == res` anyways and thus `(heap_top + incr < res) == (incr < 0)`